### PR TITLE
Add deprecation warning for `TemplatedConfigLoader` and `ConfigLoader`

### LINKED ISSUE #2693: Add deprecation warnings to `ConfigLoader` and `TemplatedConfigLoader`
## Description
We plan to deprecate `ConfigLoader` and `TemplatedConfigLoader` and replace them with the `OmegaConfigLoader` from `0.19.0` onwards. 

## Implementation
Add a deprecation warning in the code and docs. 

## Follow-up
- [ ] #2699

### PR DESCRIPTION
**Pull request by ankatiyar**
## Description

Resolve #2693 
## Development notes

Added the deprecation warning in `session.py` when the context is loaded.
### Alternate Approach
* Could also do this in the `__init__` functions of `ConfigLoader` and `TemplatedConfigLoader` but those warnings show up multiple times because the config loader is also initialised when loading logging then later when context is created.
* Used `FutureWarning` instead of `DeprecationWarning` as it is visible by default (See https://github.com/kedro-org/kedro/pull/2747 and https://github.com/kedro-org/kedro/issues/2744)
## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by SajidAlamQB — APPROVED** [review]
LGTM! 🌟 Just left some minor suggestions, thank you @ankatiyar.

File: kedro/framework/session/session.py [hunk 1]
```diff
@@ -262,7 +263,14 @@ def load_context(self) -> KedroContext:
         env = self.store.get("env")
         extra_params = self.store.get("extra_params")
         config_loader = self._get_config_loader()
-
+        if isinstance(config_loader, (ConfigLoader, TemplatedConfigLoader)):
+            warnings.warn(
+                f"{type(config_loader).__name__} will be deprecated in Kedro 0.19."
+                f" Please use the OmegaConfigLoader instead. To consult"
+                f" the documentation for OmegaConfigLoader, see here:"
+                f" https://docs.kedro.org/en/stable/configuration/advanced_configuration.html#omegaconfigloader",  # pylint: disable=line-too-long
```
[the remark this answers is not in the document]
Comment (SajidAlamQB) [reply kedro/framework/session/session.py:271]: Minor suggestions (feel free to use or ignore):

- Maybe just split the link into multiple lines.
- Add the new import line for `OmegaConfigLoader` for convenience.

**Review by merelcht — APPROVED** [review]
LGTM 👍 Don't forget to add this to the release notes. This is an important deprecation that are users should be made aware of.

File: kedro/framework/session/session.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ankatiyar) [reply kedro/framework/session/session.py]: Thanks Sajid, I split the link into multiple lines but I think the import line will make it too verbose. The link to the `OmegaConfigLoader` docs has this right on top. :)

**Pull request merged**